manifest/schema1/manifest.go

```diff
@@ -75,6 +76,9 @@ func (sm *SignedManifest) UnmarshalJSON(b []byte) error {
 		return err
 	}
 
+	if len(manifest.FSLayers) != len(manifest.History) {
```

Comment:
I don't think this should be done in the unmarshaling process. This should be done when checking verifying and validating the manifest, on upload.